tests/test_module.py

```diff
@@ -1425,6 +1426,233 @@ def test_dump_trimmed_series(self):
+                    assert expected_result[0:1] == actual_result[0:1]
+
+                    # present + past add
+                    assert r.execute_command('TS.ADD', key, 23, 23) == 23
+                    assert r.execute_command('TS.ADD', key, 15, 22) == 15
+                    assert r.execute_command('TS.ADD', key, 27, 27) == 27
+                    assert r.execute_command('TS.ADD', key, 23, 25) == 23
+                    assert r.execute_command('TS.ADD', key, 30, 30) == 30
+                    
+                    expected_result = r.execute_command('TS.RANGE', key, 0, -1, 'aggregation', agg_type, 10)
+                    actual_result = r.execute_command('TS.RANGE', agg_key, 0, -1)
+                    assert expected_result[0:3] == actual_result[0:3]
+                    assert 3 == self._get_ts_info(r, agg_key).total_samples
+                    assert 11 == self._get_ts_info(r, key).total_samples
+
+                    r.execute_command('DEL', key)
+                    r.execute_command('DEL', agg_key)
+
+    def t1est_downsampling_extensive(self):
+        with self.redis() as r:
+            key = 'tester'
+            fromTS = 10
+            toTS = 10000
+            type_list = ['', 'uncompressed']
+            for chunk_type in type_list:
+                agg_list = ['avg', 'sum', 'min', 'max', 'count', 'range', 'first', 'last', 'std.p', 'std.s', 'var.p', 'var.s'] #more
+                for agg in agg_list:
+                    agg_key = self._insert_agg_data(r, key, agg, chunk_type, fromTS, toTS)
+
+                    # sanity + check result have changed
+                    expected_result1 = r.execute_command('TS.RANGE', key, fromTS, toTS, 'aggregation', agg, 10)
+                    actual_result1 = r.execute_command('TS.RANGE', agg_key, fromTS, toTS)
+                    assert expected_result1 == actual_result1
+                    assert len(expected_result1) == 999
+
+                    for i in range(fromTS + 5, toTS - 4, 10):
+                        assert r.execute_command('TS.ADD', key, i, 42)
+
+                    expected_result2 = r.execute_command('TS.RANGE', key, fromTS, toTS, 'aggregation', agg, 10)
+                    actual_result2 = r.execute_command('TS.RANGE', agg_key, fromTS, toTS)
+                    assert expected_result2 == actual_result2
+
+                    # remove aggs with identical results
+                    compare_list = ['avg', 'sum', 'min', 'range', 'std.p', 'std.s', 'var.p', 'var.s']
+                    if agg in compare_list:
+                        assert expected_result1 != expected_result2
+                        assert actual_result1 != actual_result2
+
+                    sliced_actual_list1 = actual_result1[5:10]
+                    sliced_actual_list2 = actual_result2[5:10]
+
+                    if agg == 'avg':
+                        assert sliced_actual_list1 == [[60L, '656.5'], [70L, '756.5'], [80L, '856.5'], [90L, '956.5'], [100L, '1056.5']]
```

Comment:
I don't like this way of testing, if you know the original data and you know the aggregation function its better to write it in python and compare, I have no idea if those are the right values or not.

Reply:
This is done above in L1592
The idea here is to show the results have changed.